internal/pkg/cli/pipeline_show.go

```diff
@@ -18,14 +20,57 @@ type showPipelineVars struct {
 
 type showPipelineOpts struct {
 	showPipelineVars
+	pipelineName string
 
 	// Interfaces to dependencies
-	ws wsPipelineReader // NOTE The "show" command should probably read the pipeline.yml file, which is in my workspace directory (./ecs-project)
+	ws    wsPipelineReader
+	store storeReader
+}
+
+func newShowPipelineOpts(vars showPipelineVars) (*showPipelineOpts, error) {
+	ssmStore, err := store.New()
+	if err != nil {
+		return nil, fmt.Errorf("connect to environment datastore: %w", err)
+	}
+
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, fmt.Errorf("workspace cannot be created: %w", err)
+	}
+
+	opts := &showPipelineOpts{
+		showPipelineVars: vars,
+		ws:               ws,
+		store:            ssmStore,
+	}
+
+	return opts, nil
 }
 
 // Validate returns an error if the flag values passed by the user are invalid.
 func (o *showPipelineOpts) Validate() error {
-	// TODO Placeholder
+	if o.ProjectName() != "" {
+		if _, err := o.store.GetProject(o.ProjectName()); err != nil {
+			return err
+		}
+	}
+
+	data, err := o.ws.ReadPipelineManifest()
```

Comment:
dumb question: is `pipeline show` going to show data by querying CodePipeline or is it going to spit out stuff from the pipeline manifest?

If it's the former, I wonder if we even need to read the pipeline manifest? can we query instead `CodePipeline` resources that are tagged with `ecs-project={myProject}` to retrieve all the pipelines in the project

Reply:
We need to read the pipeline manifest to access the name of the pipeline, since it's not currently stored anywhere else. We could have the name stored in SSM like env/app names etc, but right now it's not persisted anywhere else. The name is then used to fetch the CFN stack, which is what will be called to fetch resource data.

Reply:
Sorry -- to answer your question, my idea was to query CFN, not CodePipeline, but I haven't compared what it would be like between one versus the other. Thoughts?

Reply:
Added a call to codepipeline to validate the pipeline name. Pipelinename can also be specified with a flag as well.